### PR TITLE
[Wholegraph] Build docker image for WholeGraph-GraphStorm

### PR DESCRIPTION
**Pull request by isratnisa**
Docker file to create docker image which supports NVIDIA's WholeGraph in GraphStorm. This Dockerfile should install EFA and NCCL on top of the other prerequisites required by GraphStorm.

### CODE REVIEW

File: docker/wholegraph/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,52 @@
+FROM nvidia/cuda:11.3.1-devel-ubuntu20.04
+ENV DEBIAN_FRONTEND=noninteractive
+ENV HOME=/root
+
```
Comment (classicsong) [inline docker/wholegraph/Dockerfile:4]: Where do you install wholegraph and its dependancies?

Comment (isratnisa) [reply docker/wholegraph/Dockerfile]: Pasted the wrong file. Fixed now.

File: docker/wholegraph/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ENV PATH "/opt/amazon/efa/bin:$PATH"
+
+# Install WholeGraph
+COPY wholegraph/install_wholegraph.sh install_wholegraph.sh
+RUN bash install_wholegraph.sh
+
+# Install GraphStorm
+RUN pip install --no-cache-dir boto3 'h5py>=2.10.0' scipy tqdm 'pyarrow>=3' 'transformers==4.28.1' pandas pylint scikit-learn ogb psutil
+RUN git clone https://github.com/awslabs/graphstorm -b 0.1.1
```
Comment (classicsong) [inline docker/wholegraph/Dockerfile:35]:
```suggestion
RUN git clone https://github.com/awslabs/graphstorm
```

File: docker/wholegraph/install_wholegraph.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+git clone https://github.com/fmtlib/fmt.git  /opt/fmt
+cd /opt/fmt
+git checkout 9.1.0
+mkdir build && cd build
+cmake -DCMAKE_POSITION_INDEPENDENT_CODE=TRUE ..
+make
+make install
+git clone https://github.com/gabime/spdlog.git  /opt/spdlog
```
Comment (classicsong) [inline docker/wholegraph/install_wholegraph.sh:9]: Add an empty line between 8 and 9

File: docker/wholegraph/install_wholegraph.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+cd /opt/spdlog && mkdir build && cd build
+cmake .. && make -j
+cp libspdlog.a  /usr/lib/libspdlog.a
+export PYTHON=/usr/bin/python
+cd  /opt/rapids/
+git clone https://github.com/rapidsai/wholegraph.git -b refactoring
```
Comment (classicsong) [inline docker/wholegraph/install_wholegraph.sh:15]: Add an empty line between 15 and 16.

File: docker/wholegraph/install_wholegraph.sh (hunk 3 above)
Comment (isratnisa) [reply docker/wholegraph/install_wholegraph.sh:10]: Done

File: docker/wholegraph/install_wholegraph.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+make install
+
+git clone https://github.com/gabime/spdlog.git  /opt/spdlog
+cd /opt/spdlog && mkdir build && cd build
+cmake .. && make -j
+cp libspdlog.a  /usr/lib/libspdlog.a
+export PYTHON=/usr/bin/python
+
+cd  /opt/rapids/
+git clone https://github.com/rapidsai/wholegraph.git -b refactoring
```
Comment (classicsong) [inline docker/wholegraph/install_wholegraph.sh:17]: The refactoring is for GraphStorm?

Comment (isratnisa) [reply docker/wholegraph/install_wholegraph.sh:17]: @chang-l Can we use the master now?

Comment (chang-l) [reply docker/wholegraph/install_wholegraph.sh:17]: `refactoring` is their early dev/integration branch. Now you can safely move to their release branch `branch-23.08`.

Comment (isratnisa) [reply docker/wholegraph/install_wholegraph.sh]: Fixed.

**Review by classicsong — APPROVED** [review]

**Review by classicsong — APPROVED** [review]

**Pull request merged**